### PR TITLE
Enable ChipWhisperer CW310

### PR DESCRIPTION
**Pull request by vogelpi**
This PR contains the required changes to enable SCA for OpenTitan on the ChipWhisperer CW310 FPGA board. The most notable changes are:
- Capture related parameters that need to be changed frequently (based on board and bitstream) like scope gain and number of samples we want to capture are moved to the yaml config file. Previously, we had them hardcoded in the sources.
- spiflash.py is replaced by the version from the OpenTitan repo. This version also supports both the CW305 and CW310 and contains some bugfixes. It is now vendored into ot-sca using the OpenTitan vendor tool.
- Enable support for the CW310. Currently we detect the board version based on the selected bitstream inside the yaml config file. Under the hood, the ChipWhisperer API uses the CW305 class for the CW310. This allows to reuse most of the code for the CW305. What currently doesn't work is the feature to detect if the bitstream had already been programmed. This is not dramatic but since the FPGA is much larger, loading the bitstream takes around 15 seconds which is not so nice.

I've tested these changes both on the CW305 (using the bitstreams and binaries in the tree) and the CW310. I will push a PR hopefully tomorrow to make the necessary changes in OpenTitan.

Below you can see two plots of 200 AES traces with DOM masking. Top is the CW310, bottom is the CW305. It doesn't look too bad here but when considering more traces, one sees that there is massive "background" activity that completely buries the AES traces. There will be some work needed to reduce the background noise. Disabling the entropy complex already helped a bit. 

cw310
cw305

### CODE REVIEW

File: cw/cw305/simple_capture_traces.py [hunk 1]
```diff
@@ -6,137 +6,145 @@
...
+from vendor.lowrisc_opentitan_util import cw_spiflash
+
+
+def initialize_capture(device_cfg, capture_cfg):
+    """Initialize capture."""
+    # Extract target board type from bitstream name.
+    if re.search('cw310', device_cfg['fpga_bitstream']):
+        board = 'CW310'
+    else:
+        board = 'CW305'
```
Comment (alphan) [inline cw/cw305/simple_capture_traces.py:27]: I think `board` should ideally be an enum, can you add a TODO for that?

Also:
```suggestion
    m = re.search('cw305|cw310', device_cfg['fpga_bitstream'])
    if m:
        if m.group() == 'cw305':
            board = 'CW305'
        else:
            assert m.group() == 'cw310'
            board = 'CW310'
    else:
        raise ValueError('Could not infer target board type from bitstream name')
```

**Review by alphan — APPROVED** [review]
Thanks @vogelpi. Just some nits.

File: cw/cw305/vendor/lowrisc_opentitan_util.vendor.hjson [hunk 2]
```diff
@@ -0,0 +1,16 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+{
+    name: "lowrisc_opentitan_util",
+    target_dir: "lowrisc_opentitan_util",
```
[the remark this answers is not in the document]
Comment (vogelpi) [reply cw/cw305/vendor/lowrisc_opentitan_util.vendor.hjson:6]: Done.

File: cw/cw305/util/device.py [hunk 3]
```diff
@@ -36,80 +36,92 @@ def wrapped_fn(*args, **kwargs):
...
+    def initialize_fpga(self, board, bitstream, pll_frequency):
+        """Initializes FPGA bitstream and sets PLL frequency."""
+        # Do not program the FPGA if it is already programmed.
+        # Note: Set this to True to force programming the FPGA when using a new
+        # bitstream.
+        # TODO: We should have this in the CLI.
+        force_programming = False
+        print('Connecting and loading FPGA... ', end='')
+        if board == 'CW310':
+            fpga = cw.target(None, cw.capture.targets.CW310, slurp=False)
```
[the remark this answers is not in the document]
Comment (vogelpi) [reply cw/cw305/util/device.py:58]: Unfortunately, it doesn't work either. Created an issue here #48. But I've switched to `fpga = cw.capture.targets.CW310()`.

File: cw/cw305/simple_capture_traces.py (hunk 1 above)
Comment (vogelpi) [reply cw/cw305/simple_capture_traces.py:27]: Created an issue, and switched to your suggested code. Thanks for your review @alphan !

**Pull request merged**